dev/com.ibm.ws.springboot.fat20.jms.app/src/main/java/com/ibm/ws/springboot/support/version20/test/jms/app/JmsConfig.java

```diff
@@ -19,23 +20,33 @@
 import org.springframework.jms.annotation.EnableJms;
 import org.springframework.jms.config.DefaultJmsListenerContainerFactory;
 import org.springframework.jndi.JndiTemplate;
+import org.springframework.transaction.jta.JtaTransactionManager;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.lookup.JndiDataSourceLookup;
 import javax.jms.ConnectionFactory;
 import javax.jms.Queue;
 
-
 @EnableJms
 @Configuration
 public class JmsConfig {
 
     @Bean
     public ConnectionFactory connectionFactory() throws NamingException{
-        return (ConnectionFactory) new JndiTemplate().lookup("jms/CF1");
+        return (ConnectionFactory) new JndiTemplate().lookup("jms/XA_CF1");
+    }
+    
+    @Bean
+    JtaTransactionManager transactionManager() {
+    	return new JtaTransactionManager();
```

Comment:
In Spring Framework 5 (Spring Boot 2) apps you should use the following instead so it will load the `org.springframework.transaction.jta.WebSphereUowTransactionManager` implementation.

```java
		JtaTransactionManager transactionManager = new JtaTransactionManagerFactoryBean().getObject();
		return transactionManager;
```

In Spring 6 this has been deprecated and one should use `return new JtaTransactionManager();` like you are doing.  This ends up looking for the containers transaction manager in JNDI, which now works in Liberty but for a Spring Framework 5 based application we should continue to test the `org.springframework.transaction.jta.WebSphereUowTransactionManager` integration